src/server/scripts/Shadowlands/SepulcherOfTheFirstOnes/boss_anduin_wrynn.cpp

```diff
@@ -1480,16 +1395,16 @@ struct boss_anduin_wrynn : public BossAI
 
         if (me->GetMap()->GetDifficultyID() == DIFFICULTY_MYTHIC_RAID)
         {
-            if (AchievementEntry const* mythicAnduin = sAchievementStore.LookupEntry(ACHIEVMENT_ANDUIN_MYTHIC))
+            if (AchievementEntry const* mythicAnduin = sAchievementStore.LookupEntry(ACHIEVEMENT_ANDUIN_MYTHIC))
             {
-                Map::PlayerList const& players = me->GetMap()->GetPlayers();
-                for (Map::PlayerList::const_iterator i = players.begin(); i != players.end(); ++i)
-                    if (Player* player = i->GetSource())
-                        player->CompletedAchievement(mythicAnduin);
+                me->GetMap()->DoOnPlayers([mythicAnduin](Player* player)
+                {
+                    player->CompletedAchievement(mythicAnduin);
```

Comment:
Are you sure this is required? The achievement surely uses a criteria kill credit on given difficulty. If it doesn't work, you need to check why.